### DISCUSSION TITLE
Simulation acceleration

### DISCUSSION THREAD

**Discussion by taejin-j** [accel-sim/accel-sim-framework#259] [Ideas]
I am using Accel-Sim with traces generated from an AI application. I have a kernel-list of 90k files and the simulation took roughly 20 days. I am trying to simulate a larger application and the expected time it would take is not practical. I was wondering if there was a way for simulation acceleration?

**Comment by JRPan** [comment]
The simulation is slow. We tried our best to run this as fast as possible but this is what we have. 

However, there are other techniques you can use. We have a paper PKA: https://dl.acm.org/doi/abs/10.1145/3466752.3480100
Which selects principle kernels and only simulates a subset of kernels. There are similar works, but this is what we use. Feel free to take a look. Thanks. 

I can ping the author if you have any specific questions when using PKA. 

Thanks

**Reply by taejin-j** [reply]
Thank you so much for your reply. I went through your paper and it was useful. However, my requirement is currently to execute all kernels. I notice that Accel-sim runs on a single thread. Would it be possible to make accel-sim multithreaded? I would appreciate any insights on how to get started.
